Uses singular mongo client for collector.

ptrr opened the pull request.

fniessink commented on components/shared_server_code/src/shared/initialization/database.py:
```diff
@@ -16,7 +16,7 @@
     """Returns a pymongo client."""
     database_url = os.environ.get("DATABASE_URL", url)
-    return pymongo.MongoClient(database_url)
+    return pymongo.MongoClient(database_url, serverSelectionTimeoutMS=1)
```
Timeout of 1 millisecond? Seems too low?

fniessink commented on components/collector/src/base_collectors/collector.py:
```diff
@@ -23,9 +25,14 @@
-    def __init__(self) -> None:
+    MAX_SLEEP_DURATION = int(os.environ.get("COLLECTOR_SLEEP_DURATION", 20))
+    MEASUREMENT_LIMIT = int(os.environ.get("COLLECTOR_MEASUREMENT_LIMIT", 30))
+    MEASUREMENT_FREQUENCY = int(os.environ.get("COLLECTOR_MEASUREMENT_FREQUENCY", 15 * 60))
```
These should be imported from components/collector/src/base_collectors/config.py

ptrr committed:
Uses singular mongo client

Collector uses singular datbase connection
Adds shared_server_code changes to include database parameter
Notifier uses singular database connection

fniessink committed:
Increase coverage.
